src/rax.c

```diff
@@ -154,36 +154,69 @@ static inline void raxStackFree(raxStack *ts) {
     (((n)->iskey && !(n)->isnull)*sizeof(void*)) \
 )
 
+/* Get the node auxiliary data. */
+void *raxGetData(raxNode *n) {
+    if (n->isnull) return NULL;
+    void **ndata =(void**)((char*)n+raxNodeCurrentLength(n)-sizeof(void*));
+    void *data;
+    memcpy(&data,ndata,sizeof(data));
+    return data;
+}
```

Comment:
```suggestion
void *raxGetData(raxNode *n) {
    if (n->isnull) return NULL;
    return ((char*)n+raxNodeCurrentLength(n)-sizeof(void*));
}
```

If I'm not mistaken, these lines copy `ndata` ptr to `data` ptr. No need for that?

Reply:
@tezc your approach will return the address of the memory where the data pointer is stored, but here we want to return the data at that pointer.

Reply:
I'm so dumb that I thought it was a leftover code or something.

Reply:
@skaslev why not define it in advance instead of moving forward?

Reply:
Done in [4e7b739](https://github.com/skaslev/redis/pull/2/commits/4e7b739dbd52932607624f57bc7c7ec579424628).